x/gov/keeper/keeper.go

```diff
@@ -1,268 +1,105 @@
 }
 
-// Router returns the gov keeper's router
-func (keeper Keeper) Router() *baseapp.MsgServiceRouter {
-	return keeper.router
+// DeleteVotes implements GovKeeper.
+func (keeper *Keeper) DeleteVotes(ctx sdk.Context, proposalID uint64) {
+	// Delete all votes for the proposal using collections API
+	// Votes are stored as collections.Map[collections.Pair[uint64, sdk.AccAddress], v1.Vote]
+	// We need to clear all votes with the given proposalID prefix
+	rng := collections.NewPrefixedPairRange[uint64, sdk.AccAddress](proposalID)
+	_ = keeper.Keeper.Votes.Clear(ctx, rng)
 }
 
-// LegacyRouter returns the gov keeper's legacy router
-func (keeper Keeper) LegacyRouter() v1beta1.Router {
-	return keeper.legacyRouter
-}
-
-// GetGovernanceAccount returns the governance ModuleAccount
-func (keeper Keeper) GetGovernanceAccount(ctx sdk.Context) sdk.ModuleAccountI {
-	return keeper.authKeeper.GetModuleAccount(ctx, types.ModuleName)
-}
-
-// ProposalQueues
-
-// InsertActiveProposalQueue inserts a proposalID into the active proposal queue at endTime
-func (keeper Keeper) InsertActiveProposalQueue(ctx sdk.Context, proposalID uint64, endTime time.Time) {
-	store := ctx.KVStore(keeper.storeKey)
-	bz := types.GetProposalIDBytes(proposalID)
-	store.Set(types.ActiveProposalQueueKey(proposalID, endTime), bz)
-}
-
-// RemoveFromActiveProposalQueue removes a proposalID from the Active Proposal Queue
-func (keeper Keeper) RemoveFromActiveProposalQueue(ctx sdk.Context, proposalID uint64, endTime time.Time) {
-	store := ctx.KVStore(keeper.storeKey)
-	store.Delete(types.ActiveProposalQueueKey(proposalID, endTime))
-}
-
-// InsertInactiveProposalQueue inserts a proposalID into the inactive proposal queue at endTime
-func (keeper Keeper) InsertInactiveProposalQueue(ctx sdk.Context, proposalID uint64, endTime time.Time) {
-	store := ctx.KVStore(keeper.storeKey)
-	bz := types.GetProposalIDBytes(proposalID)
-	store.Set(types.InactiveProposalQueueKey(proposalID, endTime), bz)
-}
-
-// RemoveFromInactiveProposalQueue removes a proposalID from the Inactive Proposal Queue
-func (keeper Keeper) RemoveFromInactiveProposalQueue(ctx sdk.Context, proposalID uint64, endTime time.Time) {
-	store := ctx.KVStore(keeper.storeKey)
-	store.Delete(types.InactiveProposalQueueKey(proposalID, endTime))
-}
-
-// InsertQuorumCheckQueue inserts a QuorumCheckEntry into the quorum check queue.
-func (keeper Keeper) InsertQuorumCheckQueue(ctx sdk.Context, proposalID uint64, endTime time.Time, q v1.QuorumCheckQueueEntry) {
-	store := ctx.KVStore(keeper.storeKey)
-	bz := keeper.cdc.MustMarshal(&q)
-	store.Set(types.QuorumCheckQueueKey(proposalID, endTime), bz)
+// GetProposal implements GovKeeper.
+func (keeper *Keeper) GetProposal(ctx sdk.Context, id uint64) (v1.Proposal, bool) {
+	sdkProposal, err := keeper.Keeper.Proposals.Get(ctx, id)
+	if err != nil {
+		return v1.Proposal{}, false
+	}
+	atomoneProposal := v1.ConvertSDKProposalToAtomOne(&sdkProposal)
```

Comment:
Oh I didn't realize we need to do that, this gonna bit us each time we'll bring changes to fork proposal type and forgot to apply the same change here.

I've recently discovered an interesting pattern used in `golang.org/x/tools/gopls` that might help us to not forget to apply changes from the fork proposal type.

```
// If the size of token.FileSet changes, this will fail to compile.
const delta = int64(unsafe.Sizeof(v1.Proposal{})) - int64(unsafe.Sizeof(sdkv1.Proposal{}))
var _ [-delta * delta]int
```

So currently it doesn't work because delta = -16, for some reasom the size of `sdkv1.Proposal` is 216 while the size of `v1.Proposal` is 200. 

Aren't they supposed to be the same ?